packages/block-library/src/image/image.js

```diff
@@ -142,6 +144,16 @@ export default function Image( {
 			.then( ( blob ) => setExternalBlob( blob ) );
 	}, [ id, url, isSelected, externalBlob ] );
 
+	// Focus the caption after inserting an image from the placeholder. This is
+	// done to preserve the behaviour of focussing the first tabbable element
+	// when a block is mounted. Previously, the image block would remount when
+	// the placeholder is removed. Maybe this behaviour could be removed.
+	useEffect( () => {
+		if ( url && ! prevUrl && isSelected ) {
+			captionRef.current.focus();
+		}
+	}, [ url, prevUrl ] );
```

Comment:
Why we need to introduce this particular behavior in this PR? Is there a different between the previous implementation of Block.div and useBlockEditWrapper?

Reply:
Previously we were refocussing the block using `key`. Can be done separately I guess.

https://github.com/WordPress/gutenberg/pull/23034/files/808b311228c7da55f7e4f7552133fc0be26f97c8#diff-f5fc3d5a78f58a5274bc196ebc657c5bL275

Reply:
Split in https://github.com/WordPress/gutenberg/pull/25493